File: backend/app/schemas/spot.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.domain import ORMModel


class PhotoSpotRecommendationCreate(BaseModel):
    trip_id: UUID | None = None
    trip_point_id: UUID | None = None
    name: str
    location: str
    composition: str | None = None
    best_time: str | None = None
    photo_score: float | None = None
    tips: str | None = None
    images: list[str] = Field(default_factory=list)


class PhotoSpotRecommendationUpdate(BaseModel):
    trip_point_id: UUID | None = None
    name: str | None = None
    location: str | None = None
    composition: str | None = None
    best_time: str | None = None
    photo_score: float | None = None
    tips: str | None = None
    images: list[str] | None = None


class PhotoSpotRecommendationRead(ORMModel):
    id: UUID
    trip_id: UUID
    trip_point_id: UUID | None
    name: str
    location: str
    composition: str | None
    best_time: str | None
    photo_score: float | None
    tips: str | None
    images: list[str]
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/spot.ts
/**
 * 机位推荐类型定义
 * 与后端 `app/schemas/spot.py` 对齐
 */

/** 机位推荐 */
export interface PhotoSpotRecommendation {
  id: string
  trip_id: string
  trip_point_id: string | null
  name: string
  location: string
  composition: string | null
  best_time: string | null
  photo_score: number | null
  tips: string | null
  images: string[]
  created_at: string
  updated_at: string
}

/** 机位推荐创建请求 */
export interface PhotoSpotRecommendationCreate {
  trip_id?: string | null
  trip_point_id?: string | null
  name: string
  location: string
  composition?: string | null
  best_time?: string | null
  photo_score?: number | null
  tips?: string | null
  images?: string[]
}

/** 机位推荐更新请求 */
export interface PhotoSpotRecommendationUpdate {
  trip_point_id?: string | null
  name?: string
  location?: string
  composition?: string | null
  best_time?: string | null
  photo_score?: number | null
  tips?: string | null
  images?: string[]
}
